### ISSUE TITLE
Installing VLS on Ubuntu 20.04

### ISSUE THREAD

**Issue by Vendillah** [vala-lang/vala-language-server#264]
I'm aware that there was no template to ask questions, but hopefully you're still willing to help me out.

I'd like to install VLS on an Ubuntu 20.04 derivative for vscode. The `README` redirects me to [OBS repository](https://software.opensuse.org//download.html?project=home%3APrince781&package=vala-language-server), sure. There, opening Ubuntu, are only options listed for ***x***Ubuntu and 2***1***.04+. I'm not aware of the implications installing from different flavors and versions, but I remember them being advised against.

Could I get further information on how to install the VLS on my system?

**Comment by Prince781** [comment]
VLS is no longer supported on Ubuntu 20.04 due to an outdated Vala. If you still want to use it, you need to first enable the `vala-team/daily` PPA and then build VLS from source.

**Comment by VBBr** [comment]
> And I have indeed installed `libvala-0.58` but with an additional `0`.

I think you need `libvala-0.58-dev` package

**Comment by Vendillah** [comment]
> I think you need libvala-0.58-dev package

I believe that's the last package `libvala-0.58-dev/focal` that I found running `apt list | grep libvala-`.

In #263 someone ran
```txt
apt install meson valac libjsonrpc-glib-1.0-dev libvala-0.56-dev libgee-0.8-dev scdoc build-essential
```

I haven't so far. Could it be that? I was waiting for getting clear dependency errors before installing those as I couldn't always find the packages listed at the project's `README` in my repository.

**Comment by VBBr** [comment]
> I believe that's the last package `libvala-0.58-dev/focal` that I found running `apt list | grep libvala-`.

yes, install it and try to compile again.

**Comment by Vendillah** [comment]
Aha, I'm an idiot. I forgot I had to pass `--installed` to `apt list` so I was assuming I had it installed already and was confused by your answer. So I just did that. Thank you for your patience! I will continue with the installation process.

**Comment by Vendillah** [comment]
The remaining packages _I_ had to install were
```txt
sudo apt install libjson-glib-dev
sudo apt install libgee-0.8-dev
```
Following the README, I ended up with an installation in `/usr/bin/vala-language-server` and errors highlighted in VSCode (such as "namespace Gtk could not be found"), even though I'm able to compile and run the very basic (10 lines) program. I've tried setting up a few files (that contain warnings) just like the answer to [this question](https://stackoverflow.com/questions/58751821/how-to-configure-vscode-for-gtk3-for-intellisense-build-debug-and-g), but it didn't resolve them.

I'm not sure where to find help for that problem, so it would be nice to be redirected towards people that know. Else, this issue be closed.
